### DISCUSSION TITLE
How to add image to ios project?

### DISCUSSION THREAD

**Discussion by jackie-weiwei** [axmolengine/axmol#1816] [Q&A] [answered]
i copied the image file "block.png" to the "project/Content" directory  , then i  selected "Add files to myproject" in xcocde.

next i use this code to create a sprite with "block.png":
auto block = Sprite::create("block.png"sv); 
but the block sprite is null.

i tried to create a sprite with Helloworld.png  and its ok.
I don't know what I did wrong, can you help me? Thanks

**Comment by rh101** [comment]
> i copied the image file "block.png" to the "project/Content" directory , then i selected "Add files to myproject" in xcocde.

You should not be modifying anything in Xcode. Everything should be done via CMake, so the steps are as follows:

1 - Add the resources (graphics, audio files etc.) to `<project_dir>/Content`
2 - Re-run the `axmol build -p ios -a arm64 -c` command to re-create the Xcode project. This updated project will contain references to your new resources. You need to do this every time you add new resources that you want to reference from your source code.

**Reply by jackie-weiwei** [reply]
Do I need to delete the build_ios_arm64 folder before running this command?

**Reply by rh101** [reply, marked as answer]
No, you don't need to.

If ever you have any issues with the build, delete the `build_ios_arm64/CMakeCache.txt` file first, then run `axmol build -p ios -a arm64 -c`.

**Reply by jackie-weiwei** [reply]
/Users/jackieron/Documents/05-GameDemo/tetris/proj.ios_mac/ios/targets/ios/LaunchScreen.storyboard: Failed to find or create execution context for description '<IBCocoaTouchPlatformToolDescription: 0x600002d9a460> System content for IBVisionIdiom-SevenAndLater <IBScaleFactorDeviceTypeDescription: 0x600002d99e00> scaleFactor=2x, renderMode.identifier=(null)'.

I often get this error when compiling: What is the problem?

**Reply by rh101** [reply]
> I often get this error when compiling: What is the problem?

I've never seen that error, and judging by the results from a web-search, it is unlikely to be caused by anything in Axmol.  It seems like an Xcode issue. Just do a web-search and check the recommendations by other people on how they've solved it.

**Reply by jackie-weiwei** [reply]
thank you very much!

**Reply by rh101** [reply]
> thank you very much!

Good luck with it.  If you do find a solution to that strange compilation issue, then please do put the solution in this discussion for others to see, in case they have the same problem in the future.

**Reply by jackie-weiwei** [reply]
I certainly will.

**Comment by jackie-weiwei** [comment]
You are a great expert, thank you very much

**Reply by FHBlah** [reply]
Hi Jackie, I was having the same problem (see [How do I add image, sound, or data files to my new Axmol project? #1727](https://github.com/axmolengine/axmol/discussions/1727)). When I reached out to the Axmol community, I was told that when adding .png files ***without changes to the source code***, I should add the '-f' parameter to the axmol build command as follows:

```axmol build -p ios -a arm64 -c -f```

Cheers!

**Reply by jackie-weiwei** [reply]
I'll give it a try, thank you for your guidance,
